crates/wast/src/ast/token.rs

```diff
@@ -726,6 +696,22 @@ impl Peek for LParen {
     }
 }
 
+/// A convenience type to use with [`Parser::peek`](crate::parser::Parser::peek)
+/// to see if the next token ends an s-expression.
+pub struct RParen {
```

Comment:
I think in theory this shouldn't be necessary since the parser should otherwise report eof (or so I thought), but did the need for this arise?

Reply:
It comes in handy when resolving some of the parsing ambiguities in the component-model text format.

Reply:
Are those locations where sub-parsers or eof don't work? (or places where the text format could be changed to avoid the need for such lookahead?)

Reply:
I've now reorganized the parsing to avoid the need for this, so `RParen` is now removed.